myapp/app/controllers/tasks_controller.rb

```diff
@@ -35,12 +41,19 @@ def index
   def show
     @task = Task.find_by(id: params[:id])
     return redirect_to error_path(404) if @task.nil?
-    return redirect_to error_path(401) if @task.user_id != current_user.id
+    return redirect_to error_path(401) if @task.user_id != current_user.id && (!current_user.role_moderator? and !current_user.role_admin?)
```

Comment:
[question] Why other people can't see show page?

```
@task.user_id != current_user.id
```

Reply:
It is one of requirement, I think.

> システムの要件
> ....
> ユーザ登録し、自分が登録したタスクだけを見られるようにしたい